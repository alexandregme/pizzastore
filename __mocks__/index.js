export const TOPPINGS = [{
  name:"topping 1",
  price:1,
  isChecked:true
},{
  name:"topping 2",
  price:2,
  isChecked:false
},{
  name:"topping 3",
  price:3,
  isChecked:false
},{
  name:"topping 4",
  price:4,
  isChecked:false
},{
  name:"topping 5",
  price:5,
  isChecked:false
},{
  name:"topping 6",
  price:6,
  isChecked:false
},{
  name:"topping 7",
  price:7,
  isChecked:false
},{
  name:"topping 8",
  price:8,
  isChecked:false
}];

export const DEFAULT_TOPPING = {
  name:"default topping",
  price:0,
  isChecked:false
};

export const CUSTOM_TOPPING = {
  name:"custom topping",
  price:1,
  isChecked:true
};

export const DEFAULT_PIZZA = {
  "name": "default pizza",
  "maxToppings": 0,
  "basePrice": 0,
  "totalPrice": 0,
  toppings:[]
};

export const CUSTOM_PIZZA = {
  "name": "custom pizza",
  "maxToppings": 1,
  "basePrice": 1,
  "totalPrice": 1,
  toppings:[{
    name:"topping 1",
    price:1,
    isChecked:true
  }]
};

export const PIZZAS = [{
  "name": "small",
  "maxToppings": 3,
  "basePrice": 9.89,
  toppings: TOPPINGS
},{
  "name": "medium",
  "maxToppings": 5,
  "basePrice": 10.89,
  toppings: TOPPINGS
}, {
  "name": "large",
  "maxToppings": null,
  "basePrice": 13.49,
  toppings: TOPPINGS
}];
