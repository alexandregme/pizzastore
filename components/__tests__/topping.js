import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import Topping from '../topping';
import { CUSTOM_TOPPING } from "../../__mocks__";

describe("Topping", ()=>{
  let mountedTopping;
  beforeEach(()=>{
    mountedTopping = shallow(<Topping />);
  });

  it('renders without crashing', () => {
    shallow(<Topping />);
    expect(mountedTopping).toMatchSnapshot();
  });

  it('expect to have a input of the type checkbox', () => {
    let input = mountedTopping.find('input[type="checkbox"]');
    expect(input).toHaveLength(1);
  });

  it('expect to set the label default topping', () => {
    let label = mountedTopping.find('label');
    expect(label).toHaveLength(1);
    expect(label.text()).toEqual('default topping - 0');
  });
});


describe("Topping Custom component props", ()=>{
  let mountedTopping;
  let props = {topping: CUSTOM_TOPPING};

  beforeEach(()=>{
    mountedTopping = shallow(<Topping {...props} />);
  });

  it('expect to set the label custom topping', () => {
    let label = mountedTopping.find('label');
    expect(label).toHaveLength(1);
    expect(label.text()).toEqual('custom topping - 1');
  });
});

