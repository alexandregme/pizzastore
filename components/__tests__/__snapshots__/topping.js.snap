// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topping renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Topping />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          id="id-default topping"
          type="checkbox"
        />,
        <label
          htmlFor="id-default topping"
        >
          default topping
           - 
          0
        </label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "id": "id-default topping",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "default topping",
            " - ",
            0,
          ],
          "htmlFor": "id-default topping",
        },
        "ref": null,
        "rendered": Array [
          "default topping",
          " - ",
          0,
        ],
        "type": "label",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            id="id-default topping"
            type="checkbox"
          />,
          <label
            htmlFor="id-default topping"
          >
            default topping
             - 
            0
          </label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "id": "id-default topping",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "default topping",
              " - ",
              0,
            ],
            "htmlFor": "id-default topping",
          },
          "ref": null,
          "rendered": Array [
            "default topping",
            " - ",
            0,
          ],
          "type": "label",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
